x/programs/runtime/runtime_test.go

```diff
@@ -57,3 +58,27 @@ func TestRuntimeCallProgramComplexReturn(t *testing.T) {
 	require.NoError(err)
 	require.Equal(ComplexReturn{Program: program.Address, MaxUnits: 1000}, into[ComplexReturn](result))
 }
+
+func TestContextInjection(t *testing.T) {
+	require := require.New(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	program := newTestProgram(ctx, "context_injection")
+
+	_, err := program.Call("can_change_timestamp")
+	require.NoError(err)
+	_, err = program.CallWithTimestamp(1, "can_change_timestamp")
+	require.NoError(err)
+
+	_, err = program.Call("can_change_height")
+	require.NoError(err)
+	_, err = program.CallWithHeight(1, "can_change_height")
+	require.NoError(err)
+
+	_, err = program.Call("can_change_actor")
+	require.NoError(err)
+	_, err = program.CallWithActor(codec.CreateAddress(0, ids.GenerateTestID()), "can_change_actor")
+	require.NoError(err)
```

Comment:
I think each of these should be in their own test. You should be able to instantiate the program once and then use it in each proceeding test. I've done it before but don't remember what I did.

Reply:
Do you mean that the test should be instantiated globally and that each test should call only one function ? That sounds like fixture-based testing